core/shared/src/main/scala/fs2/compression/Gzip.scala

```diff
@@ -0,0 +1,631 @@
+                    trailerChunk(2),
+                    trailerChunk(3)
+                  ) & 0xffffffff
+
+                val actualInputCrc32 = crc32 & 0xffffffff
+
+                val expectedInputSize =
+                  unsignedToLong(
+                    trailerChunk(4),
+                    trailerChunk(5),
+                    trailerChunk(6),
+                    trailerChunk(7)
+                  )
+
+                if (expectedInputCrc32 != actualInputCrc32) {
+                  F.raiseError[Unit](
+                    new ZipException(
+                      s"Content failed CRC validation: $expectedInputCrc32 != $actualInputCrc32"
+                    )
+                  )
+                } else if (expectedInputSize != actualInputSize) {
+                  F.raiseError[Unit](new ZipException("Content failed size validation"))
+                } else {
+                  F.unit
+                }
+              }
+            }
+          }
+        }
+      }
+      .flatMap(_ => Stream.empty)
+
+  private val gzipHeaderBytes = 10
+  private val gzipMagicFirstByte: Byte = 0x1f.toByte
+  private val gzipMagicSecondByte: Byte = 0x8b.toByte
+  private object gzipCompressionMethod {
+    val DEFLATE: Byte = 8.toByte // Deflater.DEFLATED.toByte
+  }
+  private object gzipFlag {
+    def apply(flags: Byte, flag: Byte): Boolean = (flags & flag) == flag
+    def apply(flags: Byte, flag: Int): Boolean = (flags & flag) == flag
+
+    def ftext(flags: Byte): Boolean = apply(flags, FTEXT)
+    def fhcrc(flags: Byte): Boolean = apply(flags, FHCRC)
+    def fextra(flags: Byte): Boolean = apply(flags, FEXTRA)
+    def fname(flags: Byte): Boolean = apply(flags, FNAME)
+    def fcomment(flags: Byte): Boolean = apply(flags, FCOMMENT)
+    def reserved5(flags: Byte): Boolean = apply(flags, RESERVED_BIT_5)
+    def reserved6(flags: Byte): Boolean = apply(flags, RESERVED_BIT_6)
+    def reserved7(flags: Byte): Boolean = apply(flags, RESERVED_BIT_7)
+
+    val FTEXT: Byte = 1
+    val FHCRC: Byte = 2
+    val FEXTRA: Byte = 4
+    val FNAME: Byte = 8
+    val FCOMMENT: Byte = 16
+    val RESERVED_BIT_5 = 32
+    val RESERVED_BIT_6 = 64
+    val RESERVED_BIT_7: Int = 128
+  }
+  private object gzipExtraFlag {
+    val DEFLATE_MAX_COMPRESSION_SLOWEST_ALGO: Byte = 2
+    val DEFLATE_FASTEST_ALGO: Byte = 4
+  }
+  private val gzipOptionalExtraFieldLengthBytes = 2
+  private val gzipHeaderCrcBytes = 2
+  private object gzipOperatingSystem {
+    val FAT_FILESYSTEM: Byte = 0
+    val AMIGA: Byte = 1
+    val VMS: Byte = 2
+    val UNIX: Byte = 3
+    val VM_CMS: Byte = 4
+    val ATARI_TOS: Byte = 5
+    val HPFS_FILESYSTEM: Byte = 6
+    val MACINTOSH: Byte = 7
+    val Z_SYSTEM: Byte = 8
+    val CP_M: Byte = 9
+    val TOPS_20: Byte = 10
+    val NTFS_FILESYSTEM: Byte = 11
+    val QDOS: Byte = 12
+    val ACORN_RISCOS: Byte = 13
+    val UNKNOWN: Byte = 255.toByte
+
+    val THIS: Byte = System.getProperty("os.name") match {
```

Comment:
We'll need to do something about this for JS :) Node has https://nodejs.org/api/process.html#processplatform.

Reply:
Done.